src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,155 @@
+package org.jenkinsci.plugins.edithelp;
+
+import java.io.*;
+
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.RootAction;
+import hudson.model.View;
+import jenkins.model.Jenkins;
+import org.apache.tools.ant.taskdefs.Recorder;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.lang.Klass;
+import sun.misc.URLClassPath;
+
+import javax.swing.*;
+import java.net.Socket;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+@Extension
+public class EditHelpRootAction implements RootAction {
+
+    //reading the catalog of helpmanager
+    private static ArrayList<File> listWithFileNames = new ArrayList<>();
+    public static void getListFiles(String str) {
```

Comment:
@oleg-nenashev Do you mean that the problem may appear when two Jenkins instances being simultaneously launched? Then I do not understand how they can interfere. Those are two different processes working on different ports. The getListFiles function is called only once at Jenkins initiation.